### PR TITLE
Support for MicroProfile context propagation properties

### LINKED ISSUE #242: Support for `MicroProfile Context Propagation` properties
See https://github.com/eclipse/microprofile-context-propagation/blob/0980f2d63a3a231c3e9d28939b14bd3c5ee7639f/spec/src/main/asciidoc/mpconfig.asciidoc

 * for `mp.context.ThreadContext.*` properties,  I suggest to get the JDT type of https://github.com/eclipse/microprofile-context-propagation/blob/4523126d165e09a231232fb073255d72dd636424/api/src/main/java/org/eclipse/microprofile/context/ThreadContext.java#L77 and loop for each method (by excluding build method) to compute mp properties. We will able to get the javadoc to have documentation.

 * for `mp.context.ManagedExecutor.*` properties I suggest to do the same thing with https://github.com/eclipse/microprofile-context-propagation/blob/4523126d165e09a231232fb073255d72dd636424/api/src/main/java/org/eclipse/microprofile/context/ManagedExecutor.java#L136

OR an another solution is to describe those properties with JSON files a static properties like we will do that in https://github.com/redhat-developer/quarkus-ls/issues/243

### PR DESCRIPTION
**Pull request by rzgry**
Fixes #242 

image
image
image

### CODE REVIEW

**Comment by angelozerr** [comment]
@rzgry could you fix the conflicts please. Thanks

**Comment by rzgry** [comment]
@angelozerr fixed

File: microprofile.jdt/com.redhat.microprofile.jdt.test/src/main/java/com/redhat/microprofile/jdt/core/PropertiesManagerClassPathKindTest.java [hunk 1]
```diff
@@ -100,14 +102,11 @@ public void configQuickstartTest() throws Exception {
...
-																												 * (src)
-																												 * Java
-																												 * sources
-																												 */
-				+ 3 /* properties from (test) Java sources */,
+		assertProperties(infoFromTest, 181 /* properties from JAR */ + //
+		 3 /* properties from JAR (test) */ +  // 
+		 3 /* properties from (src) Java sources */ + //
+		 3 /* properties from (test) Java sources */ + //
+		 7 /* static properties */,
```
Comment (angelozerr) [inline microprofile.jdt/com.redhat.microprofile.jdt.test/src/main/java/com/redhat/microprofile/jdt/core/PropertiesManagerClassPathKindTest.java:109]: `7 /* static properties from microprofile-context-propagation-api */`

File: microprofile.jdt/com.redhat.microprofile.jdt.test/src/main/java/com/redhat/microprofile/jdt/core/PropertiesManagerClassPathKindTest.java [hunk 2]
```diff
@@ -65,7 +65,9 @@ public void configQuickstartTest() throws Exception {
 				MicroProfilePropertiesScope.SOURCES_AND_DEPENDENCIES, JDT_UTILS,
 				DocumentFormat.Markdown, new NullProgressMonitor());
 		Assert.assertEquals(ClasspathKind.SRC, infoFromSrc.getClasspathKind());
-		assertProperties(infoFromSrc, 181 /* properties from JAR */ + 3 /* properties from Java sources */,
+		assertProperties(infoFromSrc, 181 /* properties from JAR */ + //
+				3 /* properties from Java sources */ + //
+				7 /* static properties */,
```
Comment (angelozerr) [inline microprofile.jdt/com.redhat.microprofile.jdt.test/src/main/java/com/redhat/microprofile/jdt/core/PropertiesManagerClassPathKindTest.java:70]: 7 /* static properties from microprofile-context-propagation-api */

File: microprofile.jdt/com.redhat.microprofile.jdt.test/src/main/java/com/redhat/microprofile/jdt/core/ConfigItemIntBoolDefaultValueTest.java [hunk 3]
```diff
@@ -32,7 +32,8 @@ public void configItemIntBoolDefaultValueTest() throws Exception {
 
 		assertProperties(infoFromClasspath, 257 /* properties from JAR */ + //
 				9 /* properties from Java sources with ConfigProperty */ + //
-				2 /* properties from Java sources with ConfigRoot */,
+				2 /* properties from Java sources with ConfigRoot */ + //
+				7 /* static properties */,
```
Comment (angelozerr) [inline microprofile.jdt/com.redhat.microprofile.jdt.test/src/main/java/com/redhat/microprofile/jdt/core/ConfigItemIntBoolDefaultValueTest.java:36]: 7 /* static properties from microprofile-context-propagation-api */

File: microprofile.jdt/com.redhat.microprofile.jdt.test/src/main/java/com/redhat/microprofile/jdt/core/PropertiesManagerClassPathKindTest.java (hunk 1 above)
Comment (rzgry) [reply microprofile.jdt/com.redhat.microprofile.jdt.test/src/main/java/com/redhat/microprofile/jdt/core/PropertiesManagerClassPathKindTest.java]: Updated

**Comment by angelozerr** [comment]
Thanks @rzgry !

**Pull request merged**